cogs/help.py

```diff
@@ -1,63 +1,185 @@
+            disable_all_buttons()
+            content = "Ok, cancelled." if confirm_view.value is False else \
+                "~~Are you really sure you want to make a help thread?~~ I guess not..."
+
+            await interaction.edit_original_message(content = content, view = confirm_view)
+        else:
+            disable_all_buttons()
+            await interaction.edit_original_message(
+                content = "Created!",
+                view = confirm_view
+            )
+            await self.create_help_thread(interaction)
 
 
 class HelpView(ui.View):
     def __init__(self):
-        super().__init__(timeout=None)
-    # Nextcord help
-    @ui.button(label="Nextcord help", style=ButtonStyle.red, custom_id="help:nextcord")
-    async def nextcord_help(self, _, interaction: Interaction):
-        await self.create_help_thread("Nextcord", interaction)
-    @ui.button(label="D.py libs help", style=ButtonStyle.gray, custom_id="help:dpylibs")
-    async def nextcord_lib_help(self, _, interaction: Interaction):
-        await self.create_help_thread("Discord.py libraries", interaction)
-    
-    # Python
-    @ui.button(label="Python help", style=ButtonStyle.green, custom_id="help:python")
-    async def python_help(self, _, interaction: Interaction):
-        await self.create_help_thread("Python", interaction)
-    async def create_help_thread(self, name, interaction):
-        thread = await interaction.channel.create_thread(name=f"{name} help ({interaction.user})", type=ChannelType.public_thread)
-        await interaction.response.send_message("Created!", ephemeral=True)
-        await thread.add_user(interaction.user)
-        log_channel = nextcord.utils.get(interaction.guild.channels, name='help_logs')
-        await log_channel.send(f"Help thread for {name} created by {interaction.user.mention}: {thread.mention}!")
-        await thread.send(f"<@&882192899519954944> Help needed!\nAlright now that we are all here to help, what do you need help with {interaction.user.mention}?")
+        super().__init__(timeout = None)
+        self.add_item(HelpButton("Nextcord", style = ButtonStyle.red, custom_id = "help:nextcord"))
+        self.add_item(HelpButton("D.py libs", style = ButtonStyle.gray, custom_id = "help:dpylibs"))
+        self.add_item(HelpButton("Python", style = ButtonStyle.green, custom_id = "help:python"))
+
+
+class ConfirmButton(ui.Button["ConfirmView"]):
+    def __init__(self, label: str, style: ButtonStyle, *, custom_id: str):
+        super().__init__(label = label, style = style, custom_id = custom_id)
+
+    async def callback(self, interaction: Interaction):
+        self.view.value = True if self.custom_id == "confirm_button" else False
+        self.view.stop()
+
+
+class ConfirmView(ui.View):
+    def __init__(self):
+        super().__init__(timeout = 10.0)
+        self.value = None
+        self.add_item(ConfirmButton("Yes", ButtonStyle.green, custom_id = "confirm_button"))
```

Comment:
Buttons for help should probably start with `help:...`

Reply:
I don't see the custom_id of any button being used where so I assumed it didn't matter, will add the prefix.